### DISCUSSION TITLE
GHA自动编译MPC-HC和MPC-BE

### DISCUSSION THREAD

**Discussion by eko5624** [hooke007/mpv_PlayKit#87] [其它播放器]
参考@dyphire的脚本搞出来的，有需要的可以fork玩玩~

https://github.com/eko5624/mpc-hc
https://github.com/eko5624/MPC-BE

**Comment by hooke007** [comment]
be官方不是每日都发便携，安装，debug，filter版本吗

**Reply by eko5624** [reply]
是的，可以拿来备用。

**Comment by dyphire** [comment]
be的字幕渲染器其实也存在这个问题 https://github.com/clsid2/mpc-hc/issues/1516 ，hc上已修复，一直没去be下反馈（

**Reply by dyphire** [reply]
在be上测试编译了下hc的修复方案，可以正常工作

**Reply by dyphire** [reply]
已提交pr合并至be主分支

**Comment by zhongfly** [comment]
最大问题在于同步仓库似乎是需要手动…我更建议直接分开，而不是fork，然后用action/checkout或者git拉取源代码

https://github.com/actions/checkout#usage

用一个专门的仓库把这些workflow，commit历史就更加清晰(一个软件用一个workflow，我自己就放了一些零碎的在https://github.com/zhongfly/Action-Build/tree/main/.github/workflows

**Reply by dyphire** [reply]
我是使用的单独构建的脚本：[mpc-be](https://github.com/dyphire/Action-Build/blob/main/.github/workflows/mpc-be.yml), [mpc-hc](https://github.com/dyphire/Action-Build/blob/main/.github/workflows/mpc-hc.yml)

不过同步仓库也是可以通过action自动执行的，参考：[dyphire/MPC-BE/autoupdatefork.yml](https://github.com/dyphire/MPC-BE/blob/master/.github/workflows/autoupdatefork.yml)

**Reply by zhongfly** [reply]
> 我是使用的单独构建的脚本：[mpc-be](https://github.com/dyphire/Action-Build/blob/main/.github/workflows/mpc-be.yml), [mpc-hc](https://github.com/dyphire/Action-Build/blob/main/.github/workflows/mpc-hc.yml)
> 
> 不过同步仓库也是可以通过action自动执行的，参考：[dyphire/MPC-BE/autoupdatefork.yml](https://github.com/dyphire/MPC-BE/blob/master/.github/workflows/autoupdatefork.yml)

我觉得这样并不好，会产生大量的无用运行记录…如果要自动更新，我更建议用https://github.com/apps/pull ，即保证时效(我估计是小时？)，也能节约action的使用

**Reply by zhongfly** [reply]
@dyphire 具体实践可以这样，单独开一个分支，设为默认分支，然后放置pull的配置文件，示例：
在无用的rm分支上放pull设置，同步master分支
https://github.com/zhongfly/ACL4SSR/blob/rm/.github/pull.yml

**Reply by dyphire** [reply]
无用的运行记录可以定期使用[clean.yml](https://github.com/dyphire/MPC-BE/blob/master/.github/workflows/clean.yml)清理，我觉得还可以。不过 https://github.com/apps/pull 确实更优雅的样子

**Reply by zhongfly** [reply]
> 无用的运行记录可以定期使用[clean.yml](https://github.com/dyphire/MPC-BE/blob/master/.github/workflows/clean.yml)清理，我觉得还可以。不过 https://github.com/apps/pull 确实更优雅的样子

不过由于github app限制，涉及workflow文件的更新需要手动同意
edit:好像现在可以了…
![Screenshot_20220217115503](https://user-images.githubusercontent.com/11155705/154402312-7d1fab0a-f950-4661-860a-ce8959dfe552.png)
